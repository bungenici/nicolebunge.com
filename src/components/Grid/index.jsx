import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.module.css';

class Grid extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
  };

  static defaultProps = {
    children: null,
    className: '',
  };

  render() {
    const { children, className } = this.props;

    return <div className={classNames(className, styles.grid)}>{children}</div>;
  }
}

export default Grid;
