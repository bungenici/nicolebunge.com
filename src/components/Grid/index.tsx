import classNames from 'classnames';
import React, { HTMLAttributes } from 'react';
import styles from './style.module.css';

export type GridProps = HTMLAttributes<HTMLDivElement>;

const Grid: React.FC<GridProps> = (props) => {
  const { className, ...otherProps } = props;

  return <div className={classNames(className, styles.grid)} {...otherProps} />;
};

export default Grid;
