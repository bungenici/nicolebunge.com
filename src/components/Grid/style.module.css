:root {
  --grid-max-width: 1080px;
}

.grid {
  margin-left: auto;
  margin-right: auto;
  max-width: var(--grid-max-width);
  width: 90%;
}
