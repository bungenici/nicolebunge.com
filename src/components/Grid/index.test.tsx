/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Grid from '.';

describe('Grid component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Grid />);

    expect(asFragment()).toMatchSnapshot();
  });
});
