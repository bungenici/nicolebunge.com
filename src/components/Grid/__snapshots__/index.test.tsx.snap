// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="grid"
  />
</DocumentFragment>
`;
